content/knowledge-base/guides/write-a-composition-function-in-go.md

```diff
@@ -0,0 +1,825 @@
+---
+title: Write a Composition Function in Go
+state: beta
+alphaVersion: "1.11"
+betaVersion: "1.14"
+weight: 80
+description: "Composition functions allow you to template resources using Go"
+---
+
+Composition functions (or just functions, for short) are custom programs that
+template Crossplane resources. Crossplane calls composition functions to
+determine what resources it should create when you create a composite resource
+(XR). Read the
+[concepts](/latest/concepts/composition-functions)
+page to learn more about composition functions.
+
+You can write a function to template resources using a general purpose
+programming language. Using a general purpose programming language allows a
+Function to use more advanced logic to template resources, like loops and
+conditionals. This guide explains how to write a composition function in
+[Go](https://go.dev).
+
+{{< hint "important" >}}
+It helps to be familiar with
+[how composition functions work](/latest/concepts/composition-functions#how-composition-functions-work)
+before following this guide.
+{{< /hint >}}
+
+## Understand the steps
+
+This guide covers writing a composition function for an
+{{<hover label="xr" line="2">}}XBuckets{{</hover>}} composite resource (XR).
```

Comment:
out of scope to show (or point to) how to create the XRD that would make this XR possible/exist?

Reply:
Yes, especially by end of day today.

Reply:
I guess we could link to some docs, but seeing as this specific guide doesn't cover actually installing the function into a control plane I'm leaning toward leaving it out. 🤔